lib/prepare_release.js

```diff
@@ -369,24 +374,35 @@ export default class ReleasePreparation {
     return missing;
   }
 
-  calculateNewVersion() {
-    let newVersion;
-
-    const lastTagVersion = semver.clean(this.getLastRef());
-    const lastTag = {
-      major: semver.major(lastTagVersion),
-      minor: semver.minor(lastTagVersion),
-      patch: semver.patch(lastTagVersion)
+  findMostRecentReleaseFromChangelog(pathToChangelog, isMainChangelog) {
+    const data = readFileSync(pathToChangelog, 'utf8');
+    const [, major, minor, patch] = (isMainChangelog
+      ? /<b><a href="doc\/changelogs\/CHANGELOG_V(\d+)\.md#\1\.(\d+)\.(\d+)">\1\.\2\.\3<\/a><\/b><br\/>/
+      : /<a href="#(\d+)\.(\d+)\.(\d+)">\1\.\2\.\3<\/a><br\/>/).exec(data);
+
+    return {
+      major,
+      minor,
+      patch,
+      tag: `v${major}.${minor}.${patch}`
     };
+  }
 
-    const changelog = this.getChangelog();
+  calculateNewVersion() {
```

Comment:
Any reason not to use `pkjs@nv` instead of match parsing the changelog? Also, note we are parsing the CHANGELOG **before** resetting/sync the branch with upstream/vX.-staging, which means it could be outdated and the "new version" already exists

Reply:
> note we are parsing the CHANGELOG **before** resetting/sync the branch with upstream/vX.-staging

It's the same time we were checking for the tag before, so I don't think it can produce different results (i.e. if you try to prepare a release from an outdated branch, you might be preparing a release that's already out, before or after this change).

> Any reason not to use `pkjs@nv` instead of match parsing the changelog?

What's that?

Reply:
Wrong name, sorry. I meant https://github.com/pkgjs/nv.

Reply:
One problem would be that it only exposes an async function – which makes sense because it does a network call to <https://nodejs.org/dist/index.json> – and we can't have an async constructor. It is technically possible to make it sync with e.g. a [synchronous worker](https://www.npmjs.com/package/synchronous-worker), not sure it's worth it though

Reply:
I refactored it to a new async `prepareLocalBranch` method, and kept the changelog parsing, but after suggesting fetching the git branch from the upstream. PTAL

Reply:
I still believe the pkgjs@nv would simplify the code, but I can see your point of view, and that's fine for me. Approved :)